test-suite/js/src/test/scala/org/scalajs/testsuite/jsinterop/NestedJSClassTest.scala

```diff
@@ -423,6 +423,29 @@ class NestedJSClassTest {
     assertEquals("InnerJSClass(5) of issue 4086", obj.toString())
   }
 
+  @Test def doublyNestedInnerObject_issue4114(): Unit = {
+    val outer1 = new DoublyNestedInnerObject_Issue4114().asInstanceOf[js.Dynamic]
+    val outer2 = new DoublyNestedInnerObject_Issue4114().asInstanceOf[js.Dynamic]
+
+    outer2.middle.inner.x = 10
+
+    assertEquals("object", js.typeOf(outer1.middle))
+    assertEquals(1, outer1.middle.inner.x)
+    assertEquals(10, outer2.middle.inner.x)
+  }
+
+  @Test def triplyNestedObject_issue4114(): Unit = {
+    val obj = TriplyNestedObject_Issue4114
```

Comment:
```suggestion
    val obj = TriplyNestedObject_Issue4114.asInstanceOf[js.Dynamic]
```
Otherwise I think you're mostly testing the Scala typechecker, rather than our own code.